pkg/operations/kubernetesupgrade/upgradeagentnode.go

```diff
@@ -80,18 +80,26 @@ func (kan *UpgradeAgentNode) CreateNode(poolName string, agentNo int) error {
 	deploymentSuffix := random.Int31()
 	deploymentName := fmt.Sprintf("agent-%s-%d", time.Now().Format("06-01-02T15.04.05"), deploymentSuffix)
 
-	_, err := kan.Client.DeployTemplate(
+	depExt, err := kan.Client.DeployTemplate(
```

Comment:
there is another possibility here. you are assuming it will work or there will be a list of deployment errors.  If the deployment fails early on, due to something like a quota issue, it will return the err in the response and won't have started any deployment operations. In the RP I have an example in arminteractor where we called deploytemplate there on how to find that error and parse it out.